shop/src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,154 @@
+
+
+
+
+    private Mapper mapper = new Mapper();
+
+    public ProductService(IProductInterfaceRepository productInterfaceRepository,ISupplierInterfaceRepository supplierInterfaceRepository,IProductCategoryInterfaceRepository productCategoryInterfaceRepository)
+    {
+        this.productInterfaceRepository=productInterfaceRepository;
+        this.productCategoryInterfaceRepository= productCategoryInterfaceRepository;
+        this.supplierInterfaceRepository=supplierInterfaceRepository;
+    }
+
+
+    public ProductCategory saveProductCategory(ProductCategory productCategory)
+    {
+        return productCategoryInterfaceRepository.save(productCategory);
+    }
+
+    public ProductCategory findProductCategoryById(Long categoryId)
+    {
+        if (productCategoryInterfaceRepository.existsById(categoryId))
+        {
+            return productCategoryInterfaceRepository.getById(categoryId);
+        }
+        else{
+            throw new NotFoundException("Category doesn't exist");
+        }
+    }
+
+    public List<ProductCategory> getAllProductsCategories()
+    {
+        return productCategoryInterfaceRepository.findAll();
+    }
+
+    public void deleteProductCategoryById(Long categoryId)
+    {
+        if (productCategoryInterfaceRepository.existsById(categoryId))
+        {
+            productCategoryInterfaceRepository.deleteById(categoryId);
+        }
+    }
+
+
+    public Supplier saveSupplier(Supplier supplier)
+    {
+        return supplierInterfaceRepository.save(supplier);
+    }
+
+    public Supplier findSupplierById(Long supplierId)
+    {
+        return supplierInterfaceRepository.getById(supplierId);
+    }
+
+    public List<Supplier> getAllSuppliers()
+    {
+        return supplierInterfaceRepository.findAll();
+    }
+
+    public void deleteSupplierById(Long supplierId)
+    {
+        if (supplierInterfaceRepository.existsById(supplierId))
+        {
+            supplierInterfaceRepository.deleteById(supplierId);
+        }
+    }
+
+
+        public List<Product> getAllProducts()
+        {
+            return productInterfaceRepository.findAll();
+        }
+        public Product findProductById(Long productId)
+        {
+            return productInterfaceRepository.getById(productId);
+        }
+
+        public void addProduct(Product product,long productCategoryId, long supplierId)
+        {
+            saveProduct(product,productCategoryId,supplierId);
+        }
+
+        public void deleteProduct(Long productId){
+            if (productInterfaceRepository.existsById(productId))
+            {
+                productInterfaceRepository.deleteById(productId);
+            }
+            else
+            {
+                throw new NotFoundException("Product not found!");
+            }
+        }
+
+        public void updateProduct(Product product,Long productCategoryId,Long supplierId,Long productId)
```

Comment:
Why do you need here the productId if you have the product? You can use the id directly from the product